airflow/exceptions.py

```diff
@@ -366,6 +370,8 @@ def __init__(
         self.kwargs = kwargs
         self.timeout = timeout
         # Check timeout type at runtime
+        if isinstance(self.timeout, (int, float)):
```

Comment:
this is just for convenience since BaseSensorOperator coalesces timeout from timedelta to seconds.